src/axelrod_dojo/archetypes/cycler.py

```diff
@@ -0,0 +1,142 @@
+
+    @staticmethod
+    def generate_random_sequence(sequence_length):
+        """
+        Generates a sequence of random moves when an instance is initialised
+
+        Parameters
+        ----------
+        sequence_length - length or random moves to generate
+
+        Returns
+        -------
+        list - a list of C & D actions: e.g. [C,C,D,D,C]
+        """
+        return list(map(axl.Action, random.randint(0, 1 + 1, (sequence_length, 1))))
+
+    def crossover(self, other_cycler):
+        """
+        creates and returns a new CyclerParams instance with a single crossover point in the middle
+
+        Parameters
+        ----------
+        other_cycler - the other cycler where we get the other half of the sequence
+
+        Returns
+        -------
+        CyclerParams
+
+        """
+        seq1 = self.get_sequence()
+        seq2 = other_cycler.get_sequence()
+
+        midpoint = int(len(seq1) / 2)
+
+        # If the length is odd, keep more from sequence 1
+        if len(seq1) % 2 == 1:
+            midpoint += 1
+
+        new_seq = seq1[:midpoint] + seq2[midpoint:]
+
+        return CyclerParams(sequence=new_seq)
+
+    def mutate(self):
+        """
+        Basic mutation which may change any random gene(s) in the sequence.
+        """
+        # if the mutation occurs
+        if random.rand() <= self.mutation_probability:
+            mutated_sequence = self.get_sequence()
+            for _ in range(self.mutation_potency):
+                index_to_change = random.randint(0, len(mutated_sequence))
+                # Mutation - change a single gene
+                if mutated_sequence[index_to_change] == C:
+                    mutated_sequence[index_to_change] = D
+                else:
+                    mutated_sequence[index_to_change] = C
+            self.sequence = mutated_sequence
+
+    def player(self):
+        """
+        Create and return a Cycler player with the sequence that has been generated with this run.
+
+        Returns
+        -------
+        Cycler(sequence)
+        """
+        return axl.Cycler(self.get_sequence_str())
+
+    def copy(self):
+        """
+        Returns a copy of the current cyclerParams
+
+        Returns
+        -------
+        CyclerParams - a separate instance copy of itself.
+        """
+        # seq length will be provided when copying
+        return CyclerParams(sequence=self.get_sequence(), mutation_probability=self.get_mutation_probability())
+
+    def get_sequence_str(self):
+        """
+        Concatenate all the actions as a string for constructing Cycler players
+
+        [C,D,D,C,D,C] -> "CDDCDC"
+        [C,C,D,C,C,C] -> "CCDCCC"
+        [D,D,D,D,D,D] -> "DDDDDD"
+
+        Returns
+        -------
+        str
+        """
+        string_sequence = ""
+        for action in self.sequence:
+            string_sequence += str(action)
+
+        return string_sequence
+
+    # Getters --------------------
+    def get_sequence(self):
```

Comment:
We don't really need all these methods.

Reply:
Woops, I see that this is what @marcharper's comment was (sorry viewing this in the overall conversation view and the code review tab confused me). :+1: